ath79: Target cleanup

981213 opened the pull request:
This PR made a cleanup for this target:

1. Remove useless phy-handle nodes
2. do a alphabetical reorder for devices
3. Removed quotes for board names because I guess we won't met special characters in board name.
4. Rename dts/image using \<manufacturer\>_\<board\> scheme

Compile test for all currently supported devices and tested on TL-WR940N v1 using the image of WR740N v2.

blogic commented:
@981213 merged, Thanks !

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -94,31 +88,23 @@ ath79_setup_interfaces()
+		ucidef_set_interfaces_wan "eth0"
 		ucidef_add_switch "switch0" \
-			"0@eth0" "3:lan:1" "5:lan:2" "4:wan"
+			"0@eth1" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
```
You are switching wan and lan eth here, is this by intention?
Would be good to mention this in the commit message. Also would be good to mention other removed ucidif_set_interfaces_*.
If this is already merged, the first question would still be relevant!

981213 replied:
Sorry I forgot to mention those after changing other things.
WR740N v4/WR741N v4 are using AR9331 and the builtin switch is connected to eth1. So this should be a mistake.

adrianschmutzler replied:
So you are saying that you changed it by intention and the new version is correct?
In this case, you should also change the target/linux/ath79/base-files/etc/board.d/01_leds, where the wan led is assigned to eth1 ...

981213 replied:
Yes.
I'll use a separated commit for this device.

blogic commented:
merged but not pushed ....

blogic commented:
and dropped from my local tree again.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_tplink_tl-archer-c7-v2.dts:
```diff
@@ -4,7 +4,7 @@
 #include <dt-bindings/input/input.h>
 
-#include "qca9558_tl-archer-c7.dtsi"
+#include "qca9558_tplink_tl-archer-c7.dtsi"
```
If we do that, the tl-archer-c7 then really is wrong, as the device is just called "Archer C7", in contrast to e.g. "TL-WR1043ND".
First time I mentioned this, the argument was that tl is to identify the tplink. But if we have tplink in the file name now, the tl should be removed for Archer devices.
However, this is not necessarily something we need to do in this patch.

981213 replied:
I'll change this also since I have renamed so many dts and I need to do a rebase anyway.

adrianschmutzler commented on target/linux/ath79/image/generic-tp-link.mk:
```diff
@@ -1,19 +1,18 @@
 include ./common-tp-link.mk
 
-define Device/tl-archer-c7-v2
+define Device/tplink_tl-archer-c7-v2
```
Same here ...

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -94,31 +88,23 @@ ath79_setup_interfaces()
-		;;
-	"buffalo,wzr-hp-g450h")
+	"tplink,tl-wr740n-v2")
+		ucidef_set_interfaces_wan "eth0"
```
(Also here: changing from ucidef_set_interfaces_lan_wan "eth1.1" "eth0" to ucidef_set_interfaces_wan "eth0" by intention?)

981213 replied:
Yes. LAN should be handled by  ucidef_add_switch. (And I forgot this also...)

981213 commented:
Rebased and updated commit messages.
PS: I made the archer-c7 rename but I didn't add SUPPORTED_DEVICES to make sysupgrade work between images with and without my commit because there is no release images for ath79.

adrianschmutzler commented:
Nice and thanks, my comments are dealt with.
However, I cannot review the whole package due to my limited understanding of the phy stuff ...

tmn505 commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -9,123 +9,110 @@ ath79_setup_interfaces()
 
 	case "$board" in
-	"etactica,eg200")
+	etactica,eg200)
```
Please move this, to keep the file consistent.

981213 replied:
Updated.

981213 replied:
Sorry I put it at wrong place again...

blogic commented:
i wont merge anything into ath79 for a few days so that this PR does not need to be rebased several times

981213 commented:
Updated:
1. Fixed the incorrect order for etactica eg200
2. Fixed two typos for TL-WR740N in my commit (I typed ucidef_set_interface s _wan :( )
3. Fixed default configuration for TL-MR3020 v1 (Broken since my last ag71xx commit. I changed dts but forgot to change default configuration.)

tmn505 commented:
Thanks for considering my comment, looks good.

karlp commented on target/linux/ath79/image/generic.mk:
```diff
@@ -13,30 +13,36 @@ define Device/avm_fritz300e
   DEVICE_TITLE := eTactica EG200
   DEVICE_PACKAGES := kmod-usb-chipidea2 kmod-ledtrig-oneshot \
 	kmod-usb-serial kmod-usb-serial-ftdi kmod-usb-storage  kmod-fs-ext4
-  SUPPORTED_DEVICES += etactica,eg200 rme-eg200
```
why are you dropping these? this is how you can do upgrades from older images that used older names, like from ar71xx to ath79 and so on.  (You're dropping them all over the place)  without these, you get image validation failures on sysupgrade.

karlp replied:
(if anything, you should be _adding_ things here, because you're renaming everything!)

adrianschmutzler replied:
The idea was that changing from ar71xx to ath79 should require a forced sysupgrade, as config has changed at several places.  I'm not a fan of this policy, because I use an OpenWRT derivative which does not copy config anyway, but this is the reason why supported_devices are removed.

blogic replied:
its a philosophical question. I am still hoping to find time to get qca8k working on the builtin switches. at that point sysupgrade would break anyhow.

karlp replied:
I would suggest that dropping all the supported devices should be an unrelated commit.  it's rather hidden at the moment inside "cleanup".

Remember, you're also breaking updates from ath79 to ath79 by your renaming here, even though it's still within master codebase.

981213 replied:
Similar things was done by @mkreslin when he made some board rename in ramips so I dropped them instead of adding SUPPORTED_DEVICES to make sysupgrade work.
And I said about this in commit description (because title is too long)
I doubt whether it's needed to drop one line using two separated commits. (1 Drop the <manufacturer>,<board> because it's automatically generated. 2 drop the one coming from ar71xx because we shouldn't allow a sysupgrade from ar71xx to ath79 with config files saved.)

981213 replied:
BTW the only broken sysupgrade from ath79 to ath79 here is Archer C7 because I changed the board name.

981213 replied:
PS: I'm not renaming everything. the dts/image renaming is only about filename, and what sysupgrade check is the first compatible string in dts. I changed the compatible string for Archer C7 and WR740N/WR741N v4. I said about the Archer C7 renaming in a separated commit. As for WR740N/WR741N, what the previous author submitted is completely broken so I don't think anyone has been using that image successfully.(board name in scripts doesn't match dts compatible so the default configuration isn't even generated.)

981213 replied:
Since it caused an argument here, I'll make an RFC patch for dropping compatible strings from ar71xx on mailing list.

981213 commented:
Rebased and added the SUPPORTED_DEVICES from ar71xx back.
Also added SUPPORTED_DEVICES for Archer C7.

adrianschmutzler commented:
Nice!

lucize commented:
@981213 I found a thing about tl-wr1043nd v1, if I enable the switch in uboot and then boot from flash everything is working, if I boot directly using tcpdump, I see only incoming packets. can you help with this thing ? but it has a realtek switch

cshoredaniel commented:
@lucize proably my fault; some bad patches reverted in ar71xx.  If this is for ath79 though I have no idea...

lucize commented:
@cshoredaniel is an older problem, it never worked, it has something to do with uboot blocking traffic until the kernel unlocks the switch, somehow ar71xx is working and ath79 is not

981213 commented:
@lucize I can't find anything wrong by simply comparing code. The only mistake I found is that reset 8 is defined as phy reset in ar9132.dtsi. But this mistake is overrided in tl-wr1043nd-v1.dts.

981213 commented:
Fixed status led for TL-WR740N v2 and added supports for all its clone found in ar71xx. Tested on TL-WR940N v1 (which is a clone of TL-WR941ND v4.)

981213 committed:
ath79: Fix support for TL-MR3020 v1

Change lan and it's LED to eth0
It's broken since c7c807cb8c3fd6538101de885f66d4681785defe
where I changed the dts but forgot to change default configurations.

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

981213 committed:
ath79: Fix support for TL-WR740N/WR741N v4

1. Swap eth0/eth1
 Both devices are using AR9331, the builtin switch on AR9331 is
 connected to gmac1 and gmac1 is named as eth1 in ath79.
 PS: gmac1 is eth0 and gmac0 is eth1 in ar71xx because of the
 reversed initialization order.
2. Fix the incorrect compatible string in dts

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

981213 committed:
ath79: Remove quotes for boardname in scripts

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

981213 committed:
ath79: Remove useless phy-handle nodes

phy-handle is used to poll link status. They are useless when
we need fixed-link on these interfaces.

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

981213 committed:
ath79: Rename dts/image using manufacturer_board scheme

This helps getting rid of SUPPORTED_DEVICES which can be
auto-generated in image/Makefile.

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

981213 committed:
ath79: Put all devices in alphabetical order in scripts

And remove specifying lan interfaces as eth1.1 because this is
handled by ucidef_add_switch.

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

981213 committed:
ath79: Rename tl-archer-c7 to archer-c7

This router is called Archer C7 and the tl was used to identify
TP-LINK. Since we have added tplink in dts/board name, the tl
prefix is useless now.

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

981213 committed:
ath79: Move wzr-hp-g450h in image/generic.mk according to alphabetical order

Signed-off-by: Chuanhong Guo <gch981213@gmail.com>

adrianschmutzler commented:
Maybe this should be merged at some point before everyone loses track of what's in it.
If the big changes regarding alphabet and dts renames are in, it will be much easier to fix other (smaller) issues based on that.

mkresin commented:
You missed the "tplink,tl-archer-c7-v2" to "tplink,archer-c7-v2" in base-files/etc/hotplug.d/firmware/11-ath10k-caldata.

Any opinion about dropping the `SUPPORTED_DEVICES += tplink,tl-archer-c7-v2`. I'm fine if a forceful sysupgrade is required for boards which don't have a release image so far. The ar71xx should have a different boardname anyway.

981213 commented:
Fixed 10-ath10k-caldata and dropped SUPPORTED_DEVICES for tl-archer-c7-v2

mkresin commented:
Merged your PR to my [staging tree](https://git.openwrt.org/?p=openwrt/staging/mkresin.git;a=summary).

I've applied the archer-c7-v2 changes by hand and added a commit message where missing. Are these changes fine with you?

981213 commented:
I'm ok with these changes.
(And I've noticed that my renaming of archer-c7 breaks alphabetical order again. Thanks for applying it manually.

mkresin commented:
Merged! Thanks a lot!

sotux commented:
Hi @mkresin, Please add DEVICE_VARS in common_netgear.mk.
Without this, the factory image may contains the wrong BOARD_ID and HW_ID
I've tested on Netgear wnr2000 v3, [here](https://github.com/sotux/openwrt/commit/857bd36dfa58b7f523487acf84c462380b74c0ce) is my commit as a reference.
